lib/boom_notifier.ex

```diff
@@ -1,16 +1,57 @@
 defmodule BoomNotifier do
   @moduledoc false
+
   # Notify the exception to all the defined notifiers
 
   alias BoomNotifier.ErrorStorage
 
+  def walkthrough_notifiers(settings, callback) do
+    case Keyword.get(settings, :notifiers) do
+      nil ->
+        with {:ok, notifier} <- Keyword.fetch(settings, :notifier),
+             {:ok, options} <- Keyword.fetch(settings, :options) do
+          callback.(notifier, options)
+        end
```

Comment:
```suggestion
        notifier = Keyword.fetch!(settings, :notifier)
        options = Keyword.fetch!(settings, :options)
        callback.(notifier, options)
```

I don't see much benefit in using the `with` here since we are not making use of the non-matching errors